R/mmrm-methods.R

```diff
@@ -227,3 +231,32 @@ print.summary.mmrm <- function(x,
   cat("\n")
   invisible(x)
 }
+
+
+#' @describeIn mmrm_methods obtain the confidence interval for the MMRM fit results.
+#' @exportS3Method
+#' @examples
+#' # Confidence Interval:
+#' confint(object)
+confint.mmrm <- function(object, parm, level = 0.95, ...) {
+  cf <- coef(object)
+  pnames <- names(cf)
+  if (missing(parm)) {
+    parm <- pnames
+  }
+  assert(
+    check_subset(parm, pnames),
+    check_integerish(parm, lower = 1L, upper = length(cf))
+  )
+  if (is.numeric(parm)) parm <- pnames[parm]
+  assert_number(level, lower = 0, upper = 1)
+  a <- (1 - level) / 2
+  pct <- paste(format(100 * c(a, 1 - a), trim = TRUE, scientific = FALSE, digits = 3), "%")
+  coef_table <- h_coef_table(object)
+  df <- coef_table[parm, "df"]
+  ses <- coef_table[parm, "Std. Error"]
+  fac <- stats::qt(a, df = df)
+  ci <- array(NA, dim = c(length(parm), 2L), dimnames = list(parm, pct))
+  ci[] <- cf[parm] + (ses * fac) %o% c(1, -1)
```

Comment:
let's avoid `%o%` here, and keep with basic operators

Reply:
`%o%` is also from base, only a shortcut for `outer`. (similar to `%*%`). but we can change that to simpler forms as this is only two cols